inject.go

```diff
@@ -69,33 +85,44 @@ func Inject(target interface{}) Option {
 	// 		target.Field(2),  // Baz io.Writer
 	// 	]
 	//
-	// As we iterate through the arguments received by the function, we can
-	// simply copy the value into the corresponding value in the targets list.
+	// As we iterate through the fields of the generated struct, we can simply
+	// copy the value into the corresponding value in the targets list.
 	targets := make([]reflect.Value, 0, t.NumField())
 
 	for i := 0; i < t.NumField(); i++ {
 		f := t.Field(i)
-		// Skip private fields.
-		if f.PkgPath != "" {
+
+		// Skip private fields
+		if f.PkgPath != "" && !f.Anonymous {
```

Comment:
Looks like encoding/json will manually check if `Name` starts with uppercase to decide if it should be skipped. We'll do the same.

https://golang.org/cl/50711